From 2b6a2ecf2dc2ae39bca7c1237700d62ae320c97a Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Thu, 14 Dec 2017 11:52:16 +0800
Subject: [PATCH 2135/2241] fix: dts: a8k: sdhci: add vqmmc regulator support

On A8K DB CP0, the vqmmc volatage is controlled by GPIO,
and fixed to be 3.3V.
According to SDIO spec, the vqmmc should be 1.8V to
support eMMC HS200 mode.
The patch added the vqmmc regulator which sets the default
voltage to 3.3V and changes to 1.8V in eMMC HS200 mode.

Change-Id: I300d11506961fcb2f56dd4104678a6c631c2364d
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/47633
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 682a8af..4e77224 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -175,6 +175,7 @@
 				bus-width = <4>;
 				non-removable;
 				pinctrl-0 = <&cp0_sdhci_pins>;
+				vqmmc-supply = <&cp0_vccq_sd0_reg>;
 			};
 			cp0_comphy: comphy {
 				skip_pcie_power_off;
@@ -224,6 +225,19 @@
 			enable-active-high;
 			gpio = <&expander0 5 GPIO_ACTIVE_HIGH>;
 		};
+
+		cp0_vccq_sd0_reg: cp0_vccq_sd0 {
+			compatible = "regulator-gpio";
+			regulator-name = "cp0-vccq-sd0";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-boot-on;
+			gpios = <&expander0 15 GPIO_ACTIVE_HIGH>;
+			gpios-states = <0>;
+			states = <1800000 0x1
+				  3300000 0x0>;
+			enable-active-high;
+		};
 	};
 
 	cp1 {
-- 
2.7.4

